pkg/kvstore/etcd.go

```diff
@@ -974,10 +983,14 @@ func (e *etcdClient) statusChecker() {
 
 		switch {
 		case consecutiveQuorumErrors > consecutiveQuorumErrorsThreshold:
-			e.latestErrorStatus = fmt.Errorf("quorum check failed %d times in a row: %s",
+			err := fmt.Errorf("quorum check failed %d times in a row: %s",
 				consecutiveQuorumErrors, quorumError)
+			e.latestErrorStatus = err
+			e.statusCheckErrors <- err
 		case len(endpoints) > 0 && ok == 0:
-			e.latestErrorStatus = fmt.Errorf("not able to connect to any etcd endpoints")
+			err := fmt.Errorf("not able to connect to any etcd endpoints")
+			e.latestErrorStatus = err
+			e.statusCheckErrors <- err
```

Comment:
[Travis](https://travis-ci.com/github/cilium/cilium/jobs/359629928) panicked here because it tried to send on a closed channel. On a quick look I couldn't find why the channel is closed, though :/.

```
...
panic: send on closed channel
goroutine 261 [running]:
github.com/cilium/cilium/pkg/kvstore.(*etcdClient).statusChecker(0xc0004fc9c0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/kvstore/etcd.go:993 +0x7ae
created by github.com/cilium/cilium/pkg/kvstore.connectEtcdClient
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/kvstore/etcd.go:656 +0x62b
FAIL	github.com/cilium/cilium/pkg/kvstore	1.980s
FAIL
Makefile:210: recipe for target 'unit-tests' failed
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

Reply:
I think it's a race condition when the etcd session closes, the channel can already be closed before the status checker is stopped.

Reply:
Resolved by closing the channel when the status checker exits.